.github/workflows/full-ci-cd.yaml

```diff
@@ -4,6 +4,10 @@ on:
   push:
     branches: [main]
 
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: true
```

Comment:
Also for our case, we don't want to cancel in progress ci/cd workflows. We want to let any previous currently running action to finish before the next merge kicks off a ci/cd. 

Does that make sense?